### PR TITLE
[risk=low][RW-5572][RW-7280] Apply eslint dot-notation where possible

### PR DESCRIPTION
**Pull request by jmthibault79**
There are about 15 instances where we can't apply the rule.  We can disable the rule for these temporarily, and investigate in follow-up PRs.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/.eslintignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.eslintrc.js
```
Comment (jmthibault79) [inline ui/.eslintignore:1]: needed because it is not Typescript

File: ui/.eslintrc.js [hunk 2]
```diff
@@ -13,11 +13,12 @@ module.exports = {
   // This will allow it to disable prettier-incompatible rules from other rulesets.
   extends: ['prettier'],
   parserOptions: {
-    ecmaVersion: 6,
-    sourceType: 'module',
     ecmaFeatures: {
       jsx: true,
     },
+    ecmaVersion: 6,
+    project: ['./tsconfig.json'],
```
Comment (jmthibault79) [inline ui/.eslintrc.js:20]: needed by @typescript-eslint/dot-notation

File: ui/src/app/utils/zendesk.ts [hunk 3]
```diff
@@ -106,7 +106,9 @@ export function openZendeskWidget(
   // this property is dynamically loaded by the Zendesk web widget snippet,
   // and can't properly be typed. If for some reason the support widget is
   // unavailable, we'll show a notice to the user.
-  if (window['zE'] == null) {
+  // eslint-disable-next-line @typescript-eslint/dot-notation
+  const zE = window['zE'];
```
Comment (jmthibault79) [inline ui/src/app/utils/zendesk.ts:110]: The comment above indicates that this string-reference is desired, so let's make an exception

File: ui/src/app/cohort-search/attributes-page/attributes-page.component.tsx [hunk 4]
```diff
@@ -473,6 +473,9 @@ export const AttributesPage = fp.flow(
             } else {
               // CAT attributes are displayed as checkboxes in the attributes form
               if (parseInt(attr.estCount, 10) > 0) {
+                // Property 'checked' does not exist on type 'CriteriaAttribute'.
+                // TODO RW-5572 confirm proper behavior and fix
+                // eslint-disable-next-line @typescript-eslint/dot-notation
```
Comment (jmthibault79) [inline ui/src/app/cohort-search/attributes-page/attributes-page.component.tsx:478]: Need to disable the rule in 10 instances.  Some of these may be intended, but some may be bugs.  Let's investigate these.

**Review by freemabd — APPROVED** [review]
LGTM

**Pull request merged**